ui/app/AppLayouts/Chat/popups/community/CreateChannelPopup.qml

```diff
@@ -274,17 +279,23 @@ StatusModal {
                     return
                 }
                 let error = "";
+                let emoji =  popup.contentItem.channelName.input.icon.emoji
+                const found = RegExp(emojiRegexStr, 'g').exec(emoji);
+                if (found) {
+                    emoji = found[1]
+                }
+
```

Comment:
hey I am not sure I understand what this part does? could you please explain why it is needed?

Reply:
I agree that it wasn't clear. Along with the rebase, I added a comment, but I'll explain here as well.

Basically, without it, it would take the full image of the icon, including all the HTML shenanigans, and put it in the DB.
First, it's heavier and uglier.
Second, it gives us less control over it, because resizing it becomes harder, since the `img` comes with `width` and `height` attributes.

So that regex extracts the UTF emoji, so we can save that as is. It's way lighter, and we can then generate the `img` tag using `Emoji.parse` and use the right size.

Reply:
aah makes sense. I dont think I have done that for wallet though, I am thinking if there is a way this is done by any one saving emoji to the db.

Reply:
that should work :)

Reply:
I am going to leave it to you if you want to do it as a part of this, approving it :)

Reply:
I added a new issue here for later: https://github.com/status-im/status-desktop/issues/5161

Reply:
thank you